packages/vats/test/upgrading/test-upgrade-contracts.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file
+ *  cribbed from packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js
+ */
+import { resolve as importMetaResolve } from 'import-meta-resolve';
+import { test as anyTest } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { buildVatController } from '@agoric/swingset-vat';
+
+/** @type {import('ava').TestFn<Awaited<ReturnType<typeof makeTestContext>>>} */
+const test = anyTest;
+
+/**
+ * WARNING: uses ambient authority
+ * of import.meta.url
+ *
+ * We aim to use ambient authority only in test.before();
+ * splitting out makeTestContext() lets us share type info.
```

Comment:
it's only used once here so "share" is a little confusing. 
```suggestion
 * splitting out makeTestContext() lets us type t.context
```

Reply:
The type is used in `test('...', async t => {...})`; i.e. it's shared between `test.before()` and `test()`. But "lets us type..." is fine by me.